### ISSUE TITLE
File Manager Caja icons shifting during multiple-icon drag-n-drop between windows 

### ISSUE THREAD

**Issue by scott092707** [swaywm/sway#6493]
### Please fill out the following:
- **Sway Version:**
sway version 1.5.1

- **Debug Log:**
https://gist.github.com/scott092707/fbbd5f44673d38a3850c6979f9b4f7f6#file-20210906-152001_sway-log-caja_icon_shift-test

- **Configuration File:**
[This is the default config for sway 1.5.1, plus just enough extra at the end to allow me to record the desktop, use PyDialog to select the recording's filename/location, and to have qtpass come up as floating and 'border csd' (this last was for the other bug that I reported today - as qtpass was not invoked, it should not be an issue)]

https://gist.githubusercontent.com/scott092707/fbbd5f44673d38a3850c6979f9b4f7f6/raw/f03b090d3d7d33b45f614e303988ca2b12a651fe/config-bugtest

- **Stack Trace:**
NA

- **Description:**
I have noticed this in my previous version (1.5), but it is still present as of 1.5.1

I take pictures during my vacation, and use two windows of my file manager Caja (from Mate - it is an offshoot of an earlier version of Nautilus that still allows one to display the file modification date along with the filename below the icons).

I use two windows to sort the photos.  I normally sort photos into directories by date/event, but for some photos the date doesn't matter, so I sort them into sub-directories of miscellaneous - birds, plants, etc.
I have the misc folder open in one window, with the sub-directories showing, so that I can just select a bunch of photos from whatever date folder I have open in the left window, and drag-n-drop them into the relevant folders in the misc window on the right.

If I select enough icons on the left, (or perhaps some that are widely-enough spaced?), when the drag-n-drop starts entering the right window, the sub-directories in the right window shift to the right as if they were trying to get away.
The amount of shift varies, apparently due to the size of the group of icons being dragged (I have had the sub-directories completely flee offscreen).
They shift back when I move the drag-n-drop back to the left window.

One thing I had not noticed before, but is visible in the video
https://youtu.be/85HB12hpt-M
is that when returning the drag-n-drop to the left window, the icons there shift sometimes a bit also...

Caja is a GTK app running under XWayland. (Since Mate is planning a move to Wayland, no doubt Caja will eventually be able to run as a Wayland app, but not so far...)

**Comment by fluix-dev** [comment]
Does this not occur in other Wayland compositors (ie. is this a wlroots bug, a sway bug, or an application bug)? If it's only Sway, it's worth trying to update to 1.6 or master.

**Comment by scott092707** [comment]
I found that I have an earlier installation available that has both Sway and LXQt, and that has Caja installed.
I booted into this install, and tested Caja under both Sway (1.5?) and LXQt (0.14.1).

Caja exhibits the bug on LXQt (and thus X) as well as Sway, so Sway/Wayland is not the source of the bug.

I also tested Nemo and PCManFm-Qt (both come up natively under Wayland),
under both Sway and LXQt, and neither exhibited the bug.

I have filed a bug report for Caja with Debian, as well as upstream under MATE.

This issue should therefore be considered done.
